code/game/vault_door.dm

```diff
@@ -27,48 +30,70 @@
 /obj/structure/vaultdoor/proc/destroy()
 	icon_state = "empty"
 	set_opacity(0)
-	src.density = 0
-	destroyed = 1
+	src.density = FALSE
+	destroyed = TRUE
 
 /obj/structure/vaultdoor/proc/open()
-	is_busy = 1
+	is_busy = TRUE
 	flick("opening", src)
 	icon_state = "open"
 	playsound(loc, 'sound/f13machines/doorgear_open.ogg', 50, 0, 10)
 	sleep(30)
 	set_opacity(0)
-	src.density = 0
-	is_busy = 0
+	src.density = FALSE
+	is_busy = FALSE
+	is_open = TRUE
 
 /obj/structure/vaultdoor/proc/close()
-	is_busy = 1
+	is_busy = TRUE
```

Comment:
Not for this pull (unless you're keen)
but we should also do a loop on everything on the door turf and gib/damage if it shuts on them, to make it a bit more risky trying to slip through as it shuts